src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java

```diff
@@ -97,23 +110,42 @@ public CallbackService(final SecurityUtils securityUtils,
     }
 
     private <T> Optional<ResponseEntity<T>> sendRequest(final String url,
+                                                        final String callbackType,
                                                         final Class<T> clazz,
+
                                                         final CallbackRequest callbackRequest) {
+        Instant startTime = null;
+        int httpStatus = 0;
         try {
             LOG.debug("Invoking callback {}", url);
             final HttpHeaders httpHeaders = new HttpHeaders();
             httpHeaders.add("Content-Type", "application/json");
+
             final HttpHeaders securityHeaders = securityUtils.authorizationHeaders();
             if (null != securityHeaders) {
                 securityHeaders.forEach((key, values) -> httpHeaders.put(key, values));
             }
             final HttpEntity requestEntity = new HttpEntity(callbackRequest, httpHeaders);
-            return Optional.ofNullable(restTemplate.exchange(url, HttpMethod.POST, requestEntity, clazz));
+
+            startTime = Instant.now();
+
+            CallbackTelemetryThreadContext.setTelemetryContext(new CallbackTelemetryContext(callbackType));
```

Comment:
better putting this on top of the method.

Reply:
i guess just before callback http call is better. I don't want to populate context if s2s call is failed.

Reply:
moving to the top of the method all computation that takes time and has potential to fail seems to be the best approach, then, doesn't it?

Reply:
done